src/sympc/tensor/mpc_tensor.py

```diff
@@ -48,15 +54,18 @@ class MPCTensor:
 
     __slots__ = {"share_ptrs", "session", "shape"}
 
+    # Used by the SyMPCTensor metaclass
+    METHODS_FORWARD = {"numel"}
+    PROPERTIES_FORWARD = {"T"}
```

Comment:
Why is there some methods or functions defined in the ShareTensor dicts ` METHODS_FORWARD` and  `PROPERTIES_FORWARD` that are not in these ones?

Reply:
What is the difference in terms of usage between theses dicts (like  `PROPERTIES_FORWARD`) and the one above (like `PROPERTIES_FORWARD_ALL_SHARES`)?

Reply:
I'm a bit confused that all this logic is not directly in the metaclass `SyMPCTensor`. I would rather control everything at this level, and given the class that implements the metaclass, I would add either the properties/methods of the ShareTensor or the ones of the MPCTensors (which might actually live in their respective class files)

Reply:
1. Depending on if it is a method or property we need to properly "hook" it in ```def hook_property``` or ```def hook_method```.
2. There are some methods for which we can get the result only by forwarding the method to only one share tensor - like ```numel```. But there are methods for which we need to send the ```method``` to all the shares, ```transpose```.
3. I added the forwarding mechanism and how it should be handled at a tensor level (not in the metaclass `SyMPCTensor`) because they all have different behaviors (see how `ShareTensor` vs `MPCTensor` deal with the hooking)